examples/pytorch/ogc/utils.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+import scipy.sparse as sp
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+
+
+def sparse_mx_to_torch_sparse_tensor(sparse_mx):
+    """Convert a scipy sparse matrix to a torch sparse tensor."""
+    sparse_mx = sparse_mx.tocoo().astype(np.float32)
+    indices = torch.from_numpy(
+        np.vstack((sparse_mx.row, sparse_mx.col)).astype(np.int64)
+    )
+    values = torch.from_numpy(sparse_mx.data)
+    shape = torch.Size(sparse_mx.shape)
+    return torch.sparse.FloatTensor(indices, values, shape)
+
+
+def symmetric_normalize_adjacency(graph):
+    """Symmetric normalize graph adjacency matrix."""
+    adj = graph.adjacency_matrix()
+    in_degs = graph.in_degrees().float()
+    in_norm = torch.pow(in_degs, -0.5).unsqueeze(-1)
+    degi = torch.diag(torch.squeeze(torch.t(in_norm)))
+    degi = sp.coo_matrix(degi.cpu()).tocsr()
+    adj = sp.csr_matrix(
+        (adj.val.cpu(), (adj.row.cpu(), adj.col.cpu())), shape=adj.shape
+    )
+    adj = degi.dot(adj.dot(degi))
+    return adj
+
+
+class LinearNeuralNetwork(nn.Module):
+    def __init__(self, nfeat, nclass, bias=True):
+        super(LinearNeuralNetwork, self).__init__()
+        self.W = nn.Linear(nfeat, nclass, bias=bias)
+
+    def forward(self, x):
+        return self.W(x)
+
+    def test(self, embeds, g):
```

Comment:
This loss function computation belongs to the training loop in terms of PyTorch's convention as it is closely related to the downstream task.  It shouldn't be in the neural network module.

Reply:
Thanks. As suggested, we have relocated the 'test' function, renamed as 'model_test' for clarity, to the 'utils.py' file as an independent function.